app/client/ui-modules/profiles/ProfilePage.tsx

```diff
@@ -32,23 +46,96 @@ export function ProfilePage(): React.JSX.Element {
+  const authUser = useAppSelector((state) => state.currentUser.authUser);
+
+  const userRole = authUser?.role;
+  const userSignUp = currentUser?.createdAt;
+
+  const formateDateToMonthYear = (date: string | undefined): string => {
+    if (!date) return "...";
+    const dateObject = new Date(date);
+
+    return dateObject.toLocaleDateString("en-AU", {
+      year: "numeric",
+      month: "long",
+    });
+  };
+
+  React.useEffect(() => {
+    const loadProfile = async () => {
+      try {
+        if (!currentUser?.profileDataId)
+          throw new Error("User is not logged in");
+
+        const data = await apiGetProfileData(currentUser.profileDataId);
+        dispatch(setProfileData(data)); //Updating redux store
+      } catch (error) {
+        console.error("Failed to retrieve profile:", error);
+      }
+    };
+    loadProfile();
+  }, [currentUser?.profileDataId]);
+
   // NEED TO MAKE A "Local profile" so that u can make multiple changes without saving until the btn is pressed
   React.useEffect(() => {
     if (isEditing) {
       setLocalProfile(profile); // clone the latest profile when edit mode begins
     }
-  }, [isEditing, profile]);
+  }, [isEditing]);
 
   // in handle save, we can add validation for fields and cancel it if fields incorrect
-  const handleSave = () => {
-    Object.entries(localProfile).forEach(([field, value]) =>
-      dispatch(
-        updateField({ field: field as keyof typeof localProfile, value })
-      )
-    );
-    dispatch(setEditing(false));
+  const handleSave = async () => {
+    try {
+      // const userId = Meteor.userId();
+      if (!currentUser?.profileDataId) throw new Error("User is not logged in");
+
+      // Send updates to the backend
+      const updatedProfile = await apiUpdateProfileData(
+        currentUser.profileDataId,
+        localProfile
+      );
+      // Update Redux store with the latest data
+      dispatch(setProfileData(updatedProfile));
+      dispatch(setEditing(false));
+    } catch (error) {
+      console.error("Failed to update profile:", error);
+    }
+  };
+
+  const roleTopNavbar = (role: Role | undefined) => {
+    switch (role) {
+      case Role.AGENT:
+        return <AgentTopNavbar onSideBarOpened={onSideBarOpened} />;
+      case Role.LANDLORD:
+        return <LandlordTopNavbar onSideBarOpened={onSideBarOpened} />;
+      case Role.TENANT:
+        return <TenantTopNavbar onSideBarOpened={onSideBarOpened} />;
+      // fallback case
+      default:
+        return <AgentTopNavbar onSideBarOpened={onSideBarOpened} />;
+    }
   };
 
+  const roleLinks = (role: Role | undefined) => {
+    switch (role) {
+      case Role.AGENT:
+        return agentDashboardLinks;
+      case Role.LANDLORD:
+        return landlordDashboardLinks;
+      case Role.TENANT:
+        return tenantDashboardLinks;
+      // fallback case
+      default:
+        return agentDashboardLinks;
+    }
+  };
```

Comment:
Again, this probably need to deleted.
from now if you are doing a fix you think its out of your ticker scope raise it to group or create a new ticket for it. so you dont have to waste your time on it. (for instance all these logics has moved to one unified place now)